plugins/statistics.koplugin/main.lua

```diff
@@ -280,16 +378,68 @@ function ReaderStatistics:getTotalStats()
            total_stats
 end
 
+
+function ReaderStatistics:getAdvStats()
+    local total_stats = {
+        {
+            _("Last week"),"",
+            callback = function()
+                UIManager:show(KeyValuePage:new{
+                    title = _("Last week"),
+                    kv_pairs = getDatesFromAll(7,"daily_weekday"),
+                })
+            end,
+        },
+        {
+            T(_"Last month by day"),"",
+            callback = function()
+                UIManager:show(KeyValuePage:new{
+                    title = _("Last month by day"),
+                    kv_pairs = getDatesFromAll(30,"daily_weekday"),
+                })
+            end,
+        },
+        {
+            T(_"Last year by day"),"",
+            callback = function()
+                UIManager:show(KeyValuePage:new{
+                    title = _("Last year by day"),
+                    kv_pairs = getDatesFromAll(365,"daily"),
+                })
+            end,
+        },
+        {
+            T(_"Last year by week"),"",
+            callback = function()
+                UIManager:show(KeyValuePage:new{
+                    title = _("Last year by week"),
+                    kv_pairs = getDatesFromAll(365,"weekly"),
+                })
+            end,
+        },
+        {
+            T(_"Last 10 years by month"),"",
+            callback = function()
+                UIManager:show(KeyValuePage:new{
+                    title = _("Last 10 years by month"),
+                    kv_pairs = getDatesFromAll(3650,"monthly"), -- last 10 years
+                })
+            end,
+        },
+    }
+
+    return T(_"Time range"), total_stats
+end
+
 function ReaderStatistics:getStatisticsFromHistory(total_stats)
     local titles = {}
     local total_books_time = 0
-    for curr_file in lfs.dir(history_dir) do
-        local path = joinPath(history_dir, curr_file)
-        if lfs.attributes(path, "mode") == "file" then
-            local book_result = self:importFromFile(history_dir, curr_file)
-            local book_stats = book_result.stats
+    for _, v in pairs(ReadHistory.hist) do
+        local book_result = DocSettings:open(v.file)
+        if book_result.data.stats ~=nil then
```

Comment:
Assign `local book_stats = book_result:readSetting('stats')` immediately after `local book_result = ...`  statement, so you can collapse the following two if blocks into one.